Kernel/Devices/DeviceManagement.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2021, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Badge.h>
+#include <AK/NonnullRefPtrVector.h>
+#include <AK/OwnPtr.h>
+#include <AK/RefPtr.h>
+#include <AK/Time.h>
+#include <AK/Types.h>
+#include <Kernel/API/KResult.h>
+#include <Kernel/API/TimePage.h>
+#include <Kernel/Arch/x86/RegisterState.h>
+#include <Kernel/Devices/Device.h>
+#include <Kernel/UnixTypes.h>
+
+namespace Kernel {
+
+class NullDevice;
+class DeviceManagement {
+    AK_MAKE_ETERNAL;
+
+public:
+    DeviceManagement();
+    static void initialize();
+    static DeviceManagement& the();
+    void attach_null_device(Device const&);
+
+    void after_inserting(Badge<Device>, Device&);
+    void before_removal(Badge<Device>, Device&);
```

Comment:
Can we rename these to be `after_inserting_device` and `before_device_removal`? These names were just copied from `Device`, where it's more obvious whats being inserted and removed.